### ISSUE TITLE
Getting Cypress QRDA Cat III validation errors.

### ISSUE THREAD

**Issue by ivaldes1** [OSEHRA/popHealth#141]
Hi, I am getting QRDA Cat III validation errors with Cypress https://cypressvalidator.healthit.gov/:

15:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'extension': [facet 'minLength'] The value '' has a length of '0'; this underruns the allowed minimum length of '1'.
15:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'extension': '' is not a valid value of the atomic type '{urn:hl7-org:v3}st'.
42:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root': 'author_root' is not a valid value of the union type '{urn:hl7-org:v3}uid'.
54:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root': 'authors_organization_root' is not a valid value of the union type '{urn:hl7-org:v3}uid'.
73:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root': 'custodian_rganization_root' is not a valid value of the union type '{urn:hl7-org:v3}uid'.
90:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root': 'legal_authenticator_root' is not a valid value of the union type '{urn:hl7-org:v3}uid'.
102:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root': 'legal_authenticator_org_root' is not a valid value of the union type '{urn:hl7-org:v3}uid'.

**Comment by adrianlfns** [comment]
Same happened to me. 
You are just missing some configuration on your pophealth installation.

See below how to make that configuration.
 https://github.com/OSEHRA/popHealth/wiki/Cypress-System-Test

**Comment by ivaldes1** [comment]
Thanks I have the file open but I am not sure what the OID value is?

Edit Config/popHealth.yml

   - cda_header:->:identifier:->:root   (must be a OID) <<-- What is OID?

**Comment by ivaldes1** [comment]
How do I obtain a OID?

**Comment by petercyli** [comment]
You can get a free oid from this organization- https://www.viathinksoft.com/freeoid/request_oid.php

**Comment by ivaldes1** [comment]
Thanks I have my OID. When I edit  Config/popHealth.yml
cda_header:->:identifier:->:root:
and add the root identifier to add the OID and extension what is the
syntax?  Single quote, double quote or no quotes for the value?

Will this work?

  cda_header:
    :identifier:
      :root: "<somenumericOID>"
      :extension: "<anything>"

**Comment by ivaldes1** [comment]
Also do I have to restart pophealth to get the popHealth.yml values into
the system?

**Comment by ivaldes1** [comment]
So I tried it with and without quotes, service apache2 restart-->loaded
Cypress deck-->export-->validate.  Vendor ID for product in the Cypress
portal is set to <myOIDnumber>.  I am down to these validation errors on
all 5 of the root values entered into Config/pophealth.yml:

42:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root':
'<myOIDnumber>' is not a valid value of the union type
'{urn:hl7-org:v3}uid'.
54:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root':
'<myOIDnumber>' is not a valid value of the union type
'{urn:hl7-org:v3}uid'.
73:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root':
'<myOIDnumber>' is not a valid value of the union type
'{urn:hl7-org:v3}uid'.
90:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root':
'<myOIDnumber>' is not a valid value of the union type
'{urn:hl7-org:v3}uid'.
102:0: ERROR: Element '{urn:hl7-org:v3}id', attribute 'root':
'<myOIDnumber>' is not a valid value of the union type
'{urn:hl7-org:v3}uid'.

file looks fine for example with this line:  <id root="<myOIDnumber>"
extension="Astro"/> giving an error.

**Comment by ivaldes1** [comment]
Answer to all is quotes not necessary and you have to prefix the OID with
the arc which in my case is  1.3.6.1.4.1 so for me the complete value
is  1.3.6.1.4.1.<myOIDnumber>

**Comment by osehr-agent** [comment]
Yes. You need to restart the Apache2 service for the new configuration to take into effect.

 

Here is a sample configuration (only the cda header section) using bogus OIDS from my demo system.  Note that the strings are all in double quotes.

 

  cda_header:

    :identifier:

      :root: "2.16.840.1.113883.3.1234.0"

      :extension: "osehraext"

    :authors:

    - :ids:

      - :root: "2.16.840.1.113883.3.1234.1"

        :extension: "osehra"

      :device:

        :name: 'PopHealth'

        :model: '5.1.0'

      :addresses: []

      :telecoms: []

      :time:

      :organization:

        :ids:

        - :root: "2.16.840.1.113883.3.1234.2"

          :extension: "osehra"

        :name: ''

    :information_recipient:

      :identifier:

        :root:

        :extension: "MIPS_INDIVIDUAL"

    :custodian:

      :ids:

      - :root: "2.16.840.1.113883.3.1234.3"

        :extension: "osehra"

      :person:

        :given: ''

        :family: ''

      :organization:

        :ids:

        - :root: "2.16.840.1.113883.3.1234.4"

          :extension: "osehra"

        :name: ''

    :legal_authenticator:

      :ids:

      - :root: "2.16.840.1.113883.3.1234.5"

        :extension: "osehra"

      :addresses: []

      :telecoms: []

      :time:

      :person:

        :given:

        :family:

      :organization:

        :ids:

        - :root: "2.16.840.1.113883.3.1234.6"

          :extension: "osehra"

        :name: ''

**Comment by ivaldes1** [comment]
Hi, I have pophealth running, have it configured with OID, have cypress
account and test decks that I am validating. I am getting no numerators, my
C1 file isn't loading into cypress, and have many questions on mechanics.
If anyone has time to run through the mechanics of doing this by goto
meeting, please contact me off-list.

-- IV

**Comment by ivaldes1** [comment]
Loading cypress deck successfully but I am getting no numerators. How do I
get numerators?

**Comment by petercyli** [comment]
Here are the steps:

 

1.	Load the Cypress test deck with a selected “Practice”. 
2.	Select the “Providers” menu item from the Administrator menu
3.	Click on the “Practice” associated with the loading of the Cypress test deck
4.	From the dashboard, select the measure associated with the Cypress test deck.  You can search for the measure by typing the measure_id, e.g., CMS9v6 in the “measure or group title” input box.
5.	Make sure that the Reporting Period matches the performance period of the test patients in the Cypress test desk.  If you don’t know what that is, you should look into the actual QRDA CAT1 .xml file.  Search for the “Reporting Period”, and you find the starting and ending date and time.

 

After step 5 if you are still getting zero numerator, please send me the test deck.  We’ll need to additional debugging to find out what when wrong.

 

BTW, try another measure.  Please verify that you are getting same results, i.e., zero numerator.

 

Peter
